src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,120 @@
-import React from 'react';
+import React, { FormEvent, useCallback, useState } from 'react';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+import { ErrorType } from '../types/ErrorType';
+import { customFetch } from '../utils/fetchClient';
+import { validateEmailAddress } from '../helper/helper';
+import { ErrorForm } from '../types/ErrorTypesForm';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post | null,
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>,
+  setErrorType: React.Dispatch<React.SetStateAction<ErrorType | null>>,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  post,
+  setComments,
+  setErrorType,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [message, setMessage] = useState('');
+  const [hasName, setHasName] = useState(true);
+  const [hasEmail, setHasEmail] = useState(true);
+  const [isEmailValid, setIsEmailValid] = useState(true);
+  const [hasComment, setHasComment] = useState(true);
+  const [isCommentAdd, setIsCommentAdd] = useState(false);
+
+  const addNewComment = useCallback(async () => {
+    setIsCommentAdd(true);
+
+    try {
+      if (post !== null) {
+        const newComment = {
+          postId: post.id,
+          name,
+          email,
+          body: message,
+        };
+
+        const fetchComment = await customFetch.addComment(newComment);
+
+        setMessage('');
+        setComments(prevComments => [...prevComments, fetchComment]);
+        setErrorType(null);
+      }
+    } catch (error) {
+      setErrorType(ErrorType.ADD);
+    } finally {
+      setIsCommentAdd(false);
+      setHasName(true);
+      setHasEmail(true);
+      setHasComment(true);
+    }
+  }, [name, email, message, post]);
+
+  const handleFormSubmit = (ev: FormEvent) => {
+    ev.preventDefault();
+
+    let isNotValid;
+
+    if (!name) {
+      setHasName(false);
+      isNotValid = true;
+    }
+
+    if (!email) {
+      setHasEmail(false);
+      isNotValid = true;
+    }
+
+    if (!validateEmailAddress(email)) {
+      setIsEmailValid(false);
+      isNotValid = true;
+    }
+
+    if (!message) {
+      setHasComment(false);
+      isNotValid = true;
+    }
+
+    if (isNotValid) {
+      return;
+    }
+
+    addNewComment();
```

Comment:
addNewComment is an async function. You should await it to handle any promise rejections.